src/main/java/models/GameEngine.java

```diff
@@ -77,26 +80,33 @@ public void initPlayer() {
 
     /**
      * Determines if p1 or p2 is going to start
-     *
-     * @param randomNr
      */
-    public void getPlayerToStart(int randomNr) {
+    public void getPlayerToStart() {
+        int randomNr = rand.nextInt(2) + 1;
         if (randomNr == 1) {
             currentPlayer = p1;
+            opponentPlayer = p2;
         } else {
             currentPlayer = p2;
+            opponentPlayer = p1;
         }
     }
 
+    public Player getOpponentPlayer(){
```

Comment:
🤒 😷     🔥 🔥 🔥 

😆 🤣

Reply:
I think you missed range values[1,2] int randomNr = rand.nextInt();

Reply:
mohammed har rätt, range saknas! :P